pycheribuild/projects/syzkaller.py

```diff
@@ -62,92 +70,134 @@ def setup_config_options(cls, **kwargs):
-            HOSTARCH="amd64",
-            TARGETARCH=self.crosscompile_target.cpu_architecture.value,
-            TARGETOS="freebsd",
-            GOROOT=self.goroot.expanduser(),
-            GOPATH=self.gopath.expanduser(),
-            CC=self.CC, CXX=self.CXX,
-            PATH=self._new_path)
+        self._setup_make_args()
+
         if self.sysgen:
             self.generate()
 
         self.make_args.set_env(CFLAGS=" ".join(cflags))
         self.run_make(parallel=False, cwd=self.gosrc)
 
     def generate(self):
-        with self.set_env(PATH=self._new_path, SOURCEDIR=self.cheribsd_dir):
+        with self.set_env(PATH=self.get_path(), SOURCEDIR=self.cheribsd_dir):
             self.run_make("extract", parallel=False, cwd=self.gosrc)
             self.run_make("generate", parallel=False, cwd=self.gosrc)
 
+    def get_install_files(self, dir):
+        file_paths = []
+        for path in dir.iterdir():
+            if path.is_dir():
+                sub_files = self.get_install_files(path)
+                dir_name = path.name
+                for s in sub_files:
+                    s[1] = Path(dir_name) / s[1]
+                    file_paths.append(s)
+            else:
+                file_paths.append([path, path.name])
+        return file_paths
+
     def install(self, **kwargs):
         # XXX-AM: should have a propert install dir configuration
-        native_build = self.source_dir / "bin"
-        mips64_build = native_build / "freebsd_mips64"
-        syz_remote_install = self.syzkaller_install_path() / "freebsd_mips64"
+        build_output_path = self.source_dir / "bin"
+        syz_remote_install = self.syzkaller_install_path()
 
         self.makedirs(syz_remote_install)
 
-        self.install_file(native_build / "syz-manager", self.syzkaller_binary(), mode=0o755)
+        self.install_file(build_output_path / "syz-manager", self.syzkaller_binary(), mode=0o755)
 
         if not self.config.pretend:
-            # mips64_build does not exist if we preted, so skip
-            for fname in os.listdir(str(mips64_build)):
-                fpath = mips64_build / fname
-                if os.path.isfile(fpath):
-                    self.install_file(fpath, syz_remote_install / fname, mode=0o755)
+            # build does not exist if we preted, so skip
+            for fpath in self.get_install_files(build_output_path):
+                if os.path.isfile(fpath[0]):
+                    self.install_file(fpath[0], syz_remote_install / fpath[1], mode=0o755)
 
     def clean(self) -> ThreadJoiner:
         self.run_cmd(["chmod", "-R", "u+w", self.build_dir])
-        self.make_args.set_env(
-            HOSTARCH="amd64",
-            TARGETARCH=self.crosscompile_target.cpu_architecture.value,
-            TARGETOS="freebsd",
-            GOROOT=self.goroot.expanduser(),
-            GOPATH=self.gopath.expanduser(),
-            CC=self.CC, CXX=self.CXX,
-            PATH=self._new_path)
+
+        self._setup_make_args()
```

Comment:
If this is called in `setup()` it should be correct in both `clean()` and `compile()`